vana/chain_manager.py

```diff
@@ -245,9 +247,18 @@ def send_transaction(self, function: ContractFunction, account: LocalAccount, va
             return tx_hash, tx_receipt
         except ContractCustomError as e:
             decoded_error = decode_custom_error(function.contract_abi, e.data)
-            vana.logging.error(f"Failed to call contract function: {decoded_error}")
+            vana.logging.error(f"Failed to write to contract function: {decoded_error}")
+        except Exception as e:
+            vana.logging.error(f"Failed to write to contract function: {e}")
+
+    def read_contract_fn(self, function: ContractFunction):
```

Comment:
Where is this called from?

Reply:
I was calling `getNextFileToVerify` from `vana-dlp-chatgpt`, which returned a cryptic smart contract error. I created this utility here so that vana-dlp-chatgpt (and other DLPs) can use it to get nicer error messages.

Reply:
Ah, got it. Is it a ChainManager concern or could it go in vana/utils somewhere?

Reply:
My thinking was that it's for interacting with the chain so I kept it here, but can move it to utils if you have a strong preference. Plus the "write contract" equivalent is just above this function.